---
title: "Explanatory analysis of a GEO dataset"
author: "Paula Balcells"
date: "May 15, 2019"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# Introduction
The goal of this practical is to learn the basics of Bioconductor and Rmarkdown. Fo rhis purpose, we are going to work on a study from the Gene Expression Omnibus Database.

The study selected is "Insights into responses of human astrocytes to H5N1 infection by transcriptional analysis" with GEO accession code GSE66597.We have used the expression matrix, the experiment description and the experimental groups from this study.

The ExpressionSet object is created using two different methodologies:

* Creating and assembling the different components
* Using GEOquery package

# Summary
The study "Insights into responses of human astrocytes to H5N1 infection by transcriptional analysis" was done using Homo Sapiens organisms and experiment type	Expression profiling by array.

The following description was provided:	
*Increasing evidence demonstrates influenza virus can not only affect the respiratory system, but also infect CNS and lead to CNS disorder and encephalopathy and encephalitis. Astrocytes are the most abundant cells in the CNS, which are capable of producing cytokines and neurotrophic factors and are essential for brain homeostasis and neuronal function. Previous studies suggested that influenza virus can infect astrocytes and induce proinflammatory cytokines response as well as apoptosis. Nevertheless, very few mechanistic data are available regarding host responses to H5N1 infection in astrocytes. In this study, a functional genomics approach was utilized to investigate comprehensive host responses to H5N1 infection in a human astrocyte cell line, U251 cells.*
 	
## Experimental design	
U251 cells were infected by H5N1 at MOI 1 or control . At time points 6, 12, and 24h, total RNA were extracted for microarray experiment. Three replicates were performed at each time point of infection and control infection.
 	
## Platform
GPL6480:	Agilent-014850 Whole Human Genome Microarray 4x44K G4112F (Probe Name version)

## Status
Public on Jan 04, 2016
 	
## Contributor(s)	
Lin X, Wang R, Sun X, Zhang J, Jin M

## Citation(s)	
Lin X, Wang R, Zhang J, Sun X et al. Insights into Human Astrocyte Response to H5N1 Infection by Microarray Analysis. Viruses 2015 May 22;7(5):2618-40. PMID: 26008703

# Data description

```{r readTargets}
targets <- read.csv("targets.txt", sep="")
expressions <- read.delim("GSE66597_expression.txt", row.names = 1)
genes <- t(expressions)[0,]
```

```{r}
info=list(myName="Paula Balcells", myLab="BDBI Omics Techniques",
myContact="paulabalcells@gmail.com.com", myTitle="Practical Exercise on ExpressionSets")
```

Expression values: matrix of 41091 rows and 18 columns.
Covariates: sampleName, Group, Replicate, Color.
Genes: identifiers of the genes in the column 0 of the expression values matrix.

```{r}
dim(expressions)
summary(expressions)
```

Plotting the data:

```{r}
boxplot(expressions, col=targets$Color)
```

Principal components analysis:

```{r}
pcs <- prcomp(expressions)
names(pcs)
barplot(pcs$sdev)
```

The top variable genes are:

```{r}
variab <- apply(expressions, 1, sd)
orderedGenes <- genes[order(variab, decreasing=TRUE)]
head(variab[order(variab, decreasing=TRUE)])
```

# ExpressionSet

The ExpressionSet class contains the following slots:

* assayData: Matrix of expression measures, genes x samples
* phenoData: Sample level covariates, instance of class annotatedDataFrame
* Annotation: Name of annotation data source (annotation package)
* featureData: Name of features = data identifiers
* experimentData: MIAME information
* notes: Any notes

## AssayData

```{r include=FALSE}
require(Biobase)
```

The assay data slot contains a matrix of with the expression values. There are as many rows as genes and as many columns as samples.

```{r}
expressions <- as.matrix(expressions)
set <- ExpressionSet(expressions)
show(set)
```

## Information about covariates

The covariates in the targets data frame need to be included in the slot featureData as an AnnotatedDataFrame.

```{r}
columnDesc <-  data.frame(labelDescription= c("Sample Name", "Group", "Replicate", "Color"))
annot <- new("AnnotatedDataFrame", data=targets, varMetadata= columnDesc)
show(annot)
```

Add the annotated data frame to the expression set.

```{r}
rownames(pData(annot))<-pData(annot)$sampleName
set <- ExpressionSet(assayData=expressions, phenoData=annot)
show(set)
```

## Information about features

The information about the genes can be stored in the featureData slot, also as an AnnotatedDataFrame.

```{r}
set <- ExpressionSet(assayData=expressions,
phenoData=annot,
featureNames =genes)
show(set)
```

## Information about the experiment

The information about the experiment can be stored in the experimentData slot, as MIAME information.

```{r}
descExp <- new("MIAME", name= info[["myName"]],
lab= info[["myLab"]],
contact= info[["myContact"]] ,
title=info[["myTitle"]])

set <- ExpressionSet(assayData=expressions, phenoData=annot, fetureNames=genes, experimentData=descExp)

show(set)
```

## Objects of class ExpressionSet

Data in an ExpressionSet can be managed for different purposes, with consistency within the class ensured. It is possible to acces slot values as well as subsetting the ExpressionSet, for example.

```{r}
dim(exprs(set))
class(phenoData(set))
class(pData(phenoData(set)))
head(pData(phenoData(set)))
head(pData(set))
```

### Subsetting ExpressionSet

We are going to subset the ExpressionSet into the different groups of the experiment accounting for the targets covariates.

```{r}
con6hSet <- set[,pData(set)$Group=="con6h"]
con12hSet <- set[,pData(set)$Group=="con12h"]
con24hSet <- set[,pData(set)$Group=="con24h"]
hm6hSet <- set[,pData(set)$Group=="hm6h"]
hm12hSet <- set[,pData(set)$Group=="hm12h"]
hm24hSet <- set[,pData(set)$Group=="hm24h"]

dim(exprs(con6hSet))
head(pData(con6hSet))

all(colnames(exprs(con6hSet))==rownames(pData(con6hSet)))
all(colnames(exprs(hm6hSet))==rownames(pData(hm6hSet)))
```

# Data grom GEO

The GEO set that has been studied along the report corresponds to accession code in NCBI GEO GSE66597.

```{r include=FALSE}
require(GEOquery)
```

We can use the package GEOquery to retrieve the data.

```{r echo = T, results = 'hide'}
gse <- getGEO("GSE66597")
```

We can access the ExpressionSet object from the "gse" list.

```{r}
class(gse)
names(gse)
setFromGeo <- gse[[1]]
class(setFromGeo)
```

# Github repository
https://github.com/balpaula/Balcells_Paula-OmicsTechniques
